Packages/com.unity.inputsystem/InputSystem/Plugins/OnScreen/OnScreenDeviceManager.cs

```diff
@@ -0,0 +1,43 @@
+using UnityEngine.Experimental.Input.LowLevel;
+
+namespace UnityEngine.Experimental.Input.Plugins.OnScreen
+{
+    public class OnScreenDeviceManager
+    {
+        static OnScreenDeviceManager s_Instance;
+
+        public static OnScreenDeviceManager GetOnScreenDeviceManager()
+        {
+            if (s_Instance == null)
+                s_Instance = new OnScreenDeviceManager();
+
+            return s_Instance;
+        }
+
+        public InputEventPtr GetInputEventPtrForDevice(InputDevice device)
+        {
+            InputEventPtr eventPtr;
+            var buffer = StateEvent.From(device, out eventPtr);
+            return eventPtr;
+        }
+
+        public InputControl SetupInputControl(string controlPath)
+        {
+            var layout = InputControlPath.TryGetDeviceLayout(controlPath);
+            var device = InputSystem.TryGetDevice(layout);
```

Comment:
It should always create a device for on-screen devices and not feed against devices added from elsewhere. In here, see if we already have an on-screen device of the given type. If so, use it. If not create one and retain the a state buffer for it.